### PR TITLE
Integrate with SyliusShopApiPlugin

### PR DESCRIPTION
**Pull request by pamil**

### CODE REVIEW

File: tests/Application/app/config/config.yml [hunk 1]
```diff
@@ -37,6 +38,11 @@ doctrine:
 
 fos_rest:
     exception: ~
+    format_listener:
+        rules:
+            - { path: '^/shop-api', priorities: ['json'], fallback_format: json, prefer_extension: true } # <-- Add this
```
Comment (lchrusciel) [inline tests/Application/app/config/config.yml]: `# <-- Add this` probably not needed :)

File: tests/Responses/Expected/product/product_with_variant_details_page.json [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                    "path": "\/uo\/samll-tshirt.jpg"
+                }
+            ],
+            "appliedPromotions": [
+                {
+                    "code": "cp1",
+                    "decorations": [
+                        {
+                            "type": "message",
+                            "configuration": "{\"message\":\"Wololo\",\"activeOnProductDisplayPage\":true,\"activeOnProductListingPage\":false,\"activeOnCheckoutPage\":true}"
```
Comment (lchrusciel) [inline tests/Responses/Expected/product/product_with_variant_details_page.json]: We should not put this array into string

File: tests/Responses/Expected/elastic_search/product_list_page.json [hunk 3]
```diff
@@ -13,12 +13,12 @@
...
                     },
-                    "images": [],
-                    "appliedPromotions": []
+                    "images": "@...@",
+                    "appliedPromotions": "@...@"
                 }
             ],
-            "attributes": [],
-            "images": [],
+            "attributes": "@...@",
```
Comment (lchrusciel) [inline tests/Responses/Expected/elastic_search/product_list_page.json]: Shouldn't we revert this mather?

File: tests/Responses/Expected/elastic_search/product_list_page.json [hunk 4]
```diff
@@ -13,12 +13,12 @@
                         "current": 1999,
                         "currency": "GBP"
                     },
-                    "images": [],
-                    "appliedPromotions": []
+                    "images": "@...@",
+                    "appliedPromotions": "@...@"
```
Comment (lchrusciel) [inline tests/Responses/Expected/elastic_search/product_list_page.json]: Are promotions presented on product list page?

Comment (pamil) [inline tests/Responses/Expected/elastic_search/product_list_page.json]: Catalog promotions are.

File: tests/Responses/Expected/elastic_search/product_list_page.json (hunk 3 above)
Comment (pamil) [inline tests/Responses/Expected/elastic_search/product_list_page.json]: Good catch! 👍

File: tests/Responses/Expected/elastic_search/product_list_page.json [hunk 5]
```diff
@@ -13,12 +13,12 @@
                         "current": 1999,
                         "currency": "GBP"
                     },
-                    "images": [],
-                    "appliedPromotions": []
+                    "images": "@array@",
+                    "appliedPromotions": "@array@"
```
Comment (lchrusciel) [inline tests/Responses/Expected/elastic_search/product_list_page.json]: We should assert explicit value for `appliedPromotions`

File: composer.json [hunk 6]
```diff
@@ -7,6 +7,7 @@
         "php": "^7.1",
 
         "sylius/elastic-search-plugin": "^1.0@alpha",
+        "sylius/shop-api-plugin": "^1.0@beta",
```
Comment (lchrusciel) [inline composer.json]: Shouldn't we add it to suggest? It should be loosely coupled AFAIR

**Pull request merged**